test/factories/host_related.rb

```diff
@@ -167,9 +180,20 @@ def set_nic_attributes(host, attributes, evaluator)
     end
 
     trait :on_compute_resource do
+      transient do
+        expect_queue_compute true
+      end
       uuid Foreman.uuid
       association :compute_resource, :factory => :ec2_cr
-      after(:build) { |host| host.expects(:queue_compute) }
+      after(:build) { |host, evaluator| host.expects(:queue_compute) if evaluator.expect_queue_compute }
+    end
+
+    trait :with_compute_profile do
+      after(:build) do |host|
+        host.compute_profile = FactoryGirl.create(:compute_profile)
+        host.compute_profile.compute_attributes << FactoryGirl.build(:compute_attribute, :compute_resource => host.compute_resource)
```

Comment:
Should this be in the :compute_profile create call above?  e.g. a trait of the profile?